int64 current_angle
---
int64 next_angle
